### PR TITLE
Release v0.21.4

### PR DESCRIPTION
**Pull request by odow**
Closes https://github.com/jump-dev/JuMP.jl/issues/2326

We should get better at having more frequent patch releases.

### CODE REVIEW

**Comment by mlubin** [comment]
There's lots of good stuff in this release!
The PR numbers without links don't do much to help casual readers understand the changes. As a user, how does the text ``"#2300"`` help me understand how to use the new feature to query constraint conflicts?

**Comment by odow** [comment]
I was just following the layout of previous NEWS entries.

Given that Tagbot now populates a news for each release (https://github.com/jump-dev/JuMP.jl/releases/tag/v0.21.3), I'm not sure these are required anymore. It would certainly allow us to set a faster release pace.

**Comment by mlubin** [comment]
Maybe worth discussing during the next developer call. As is, I think a link to https://github.com/jump-dev/JuMP.jl/releases/tag/v0.21.4 would be more helpful than the current description with PR numbers listed as text.
Is updating NEWS really slowing down the release pace? A release like this deserves a discourse post (if not a blog post on the new jump blog), so even if we skipped NEWS we wouldn't be freed from the obligation of explaining what's in the release.

**Comment by odow** [comment]
> Is updating NEWS really slowing down the release pace?

It depends if we want to move to a rolling release, where we tag a release every PR, or say once a week if there have been new PRs. At present it's a bit ad-hoc.

**Comment by blegat** [comment]
I agree that the current NEWS is redundant with what TagBot does.

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,33 @@
...
+  * Don't export private macros
+  * Fix invalid assertion in nonlinear
+- Error if constraint has `NaN` right-hand side
+- Improve speed of tests
+  * Lots of work modularizing files in `/test`
+- Improve line numbers in macro error messages
+- Print nonlinear subexpressions
+- Various documentation updates
+- Dependency updates:
+  * Datastructures 0.18(#2315)
```
Comment (mlubin) [inline NEWS.md:27]: Straggling PR number

**Review by mlubin — APPROVED** [review]

**Comment by dourouc05** [comment]
The tagbot-generated content is not really redundant with `NEWS`: the issue and PR names are not always easy to understand for a newcomer. The list is also sometimes misleading: the issue #320 is just closed, it's far from solved, even though the generated list seems to indicate that the issue is no more present. 

Maybe it would help to have time-based releases, for instance every month? Maybe the person that merges code into JuMP would be responsible to update `NEWS` (if time allows) or to create an issue so that someone else knows that an update is required?

**Comment by odow** [comment]
Agree on the time-based releases. We plan to discuss it on the next monthly call.

**Pull request merged**